content/general/incoming-webhooks.textile

```diff
@@ -0,0 +1,104 @@
+
+Ably also supports "outgoing webhooks":/general/events. These are webhooks that can be generated by Ably to integrate Ably with other services such as "Zapier":/general/events/zapier/.
+
+h2(#configure). Configuring your incoming webhook
+
+You configure your incoming webhooks in the @Settings@ tab of the Ably "dashboard":https://www.ably.com/dashboard:
+
+1. In the @Incoming Webhooks Setup@ section, click the @Register a new webhook endpoint@ button.
+2. Give your webhook a friendly name.
+3. Specify a new or existing Ably channel which is going to receive the incoming webhooks.
+4. Click the @Generate a URL@ button.
+5. Copy the generated webhook URL to the clipboard, and use it to configure your service as required.
+
+You can optionally test your webhook is correctly configured using the Curl requests provided in the "dashboard":https://www.ably.com/dashboard.
+
+h2(#test). Test your incoming webhook
+
+You can now optionally test your incoming webhook with a Curl command such as the following:
+
+```[sh]
+curl -X POST 'https://rest.ably.io/channels/webhook-test/messages?key={{API_KEY_NAME}}:{{API_KEY_SECRET}}&enveloped=false' \
+     -H 'content-type: application/json' --data '{"some":"json"}'
+```
+
+Ably responds with the @channel@ and @messageId@ of the message that results from the inbound webhook: 
+
+```[json]
+{
+	"channel": "webhook-test",
+	"messageId": "4RcVBJsAbC:0"
+}
+```
+
+h2(#receive). Receiving the incoming webhook
+
+You can receive the incoming webhook as you would a normal Ably message on a channel. For example, if the incoming webhook data is in JSON format, you can display the data using the following code:
+
+```[nodejs]
+var ably = new require("ably").Realtime('{{API_KEY_NAME}}:{{API_KEY_SECRET}}');
+
+/* Obtain the channel that has been configured to receive webhooks */
+var channel = ably.channels.get('webhook-test');
+
+/* Incoming webhook messages can be displayed using this function */
+channel.subscribe(function(message) {
+    console.log("webhook: " + JSON.stringify(message.data));
+});
+```
+
+h2(#headers). Specifying optional headers
+
+The request body of incoming webhooks is treated as a message to be published. You can optionally set headers and parameters in your external service, if the service supports this, as described in the "REST API":/rest-api#unenveloped documentation.
+
+The following example shows how the message name could be set via a request header:
+
+```[sh]
+curl -X POST 'https://rest.ably.io/channels/webhook-test/messages?key=key:secret&enveloped=false' \
+     -H 'content-type: application/json' --data '{"some":"json"}' \
+     -H 'X-Ably-Name: webhook-message'
+```
+
+Your code to receive the named message is:
+
+```[nodejs]
+channel.subscribe('webhook-message', function(message) {
+    console.log("webhook: " + JSON.stringify(message.data));
+});
+```
+
+h2(#summary). Summary
```

Comment:
I think this is missing some more general information about how these requests will be handled. For example, what happens if it succeeds, and what happens if it fails?

I think it's also worth saying generally that incoming webhooks are just REST publishes, and so they will reflect the behaviour of the REST publish API. (I realise there's a link to that documentation but I think it should make that connection explicitly.) Therefore a successful publish will respond with a 201 statusCode; failure will result in a 4xx or 5xx code and an `ErrorInfo` response body.

Rate-limiting will be applied as with REST publishes.

Idempotent publishing is possible by including a unique message identifier in an `X-Ably-MessageId` header.